lib/system.nim

```diff
@@ -192,6 +192,13 @@ else:
 
 proc `addr`*[T](x: var T): ptr T {.magic: "Addr", noSideEffect.} =
   ## Builtin `addr` operator for taking the address of a memory location.
+  ## This works even for `let` variables or parameters
+  ## for better interop with C.
+  ##
+  ## .. note:: When you use it to write a wrapper for a C library, you should
+  ## always check that the original library does never write to data behind the
+  ## pointer that is returned from this procedure.
+  ##
```

Comment:
The note is a bit confusing in the sense that it should refer to the part about `let` variables. Read without seeing these additions here, it may sound like one should make sure that `addr` is also never used to have a C library write to a `var` variable.

Reply:
pre-existing, can be removed later.